app/Http/Controllers/StandController.php

```diff
@@ -57,7 +62,58 @@ public function createStandAssignment(Request $request): JsonResponse
 
     public function deleteStandAssignment(string $callsign): JsonResponse
     {
-        $this->standService->deleteStandAssignment($callsign);
+        $this->standService->deleteStandAssignmentByCallsign($callsign);
         return response()->json([], 204);
     }
+
+    public function getAirfieldStandStatus(Request $request): JsonResponse
+    {
+        if (!$airfield = Airfield::where('code', $request->query('airfield'))->first()) {
+            return response()->json([], 404);
+        }
+
+        return response()->json($this->getAirfieldStandStatusData($airfield));
+    }
+
+    private function getAirfieldStandStatusData(Airfield $airfield): array
+    {
+        if ($cachedResponse = Cache::get($this->getStandStatusCacheKey($airfield))) {
+            return $cachedResponse;
+        }
+
+        $standStatuses = $this->standService->getAirfieldStandStatus($airfield->code);
+        $response = [
+            'stands' => $standStatuses,
+            'generated_at' => Carbon::now()->toIso8601String(),
+            'refresh_interval_minutes' => self::AIRFIELD_STAND_STATUS_CACHE_MINUTES,
+            'refresh_at' => $this->getStandStatusRefreshTime()->toIso8601String(),
+        ];
+        Cache::put(
+            $this->getStandStatusCacheKey($airfield),
+            $response,
+            $this->getStandStatusRefreshTime()
+        );
+        return $response;
+    }
```

Comment:
This could be simplified using Cache::remember. Remember accepts a callback and if the cache key exists, that gets returned, else the callback runs and stores the response in the cache too.

Reply:
Good call, have done this